application/AuthTask.cpp

```diff
@@ -0,0 +1,76 @@
+// Licensed under the Apache-2.0 license. See README.md for details.
+
+#include "AuthTask.h"
+
+#include "auth/AccountModel.h"
+#include "auth/BaseAccount.h"
+#include "BaseInstance.h"
+#include "dialogs/AccountLoginDialog.h"
+#include "dialogs/AccountsDialog.h"
+#include "Exception.h"
+#include "MultiMC.h"
+
+AuthTask::AuthTask(const QString &type, InstancePtr instance, SessionPtr session, QObject *parent)
+	: StandardTask(parent), m_type(type), m_instance(instance), m_session(session)
+{
+}
+
+void AuthTask::executeTask()
+{
+	BaseAccount *account = MMC->accountsModel()->get(m_type, m_instance);
+	if (account)
+	{
+		try
+		{
+			setStatus(tr("Checking if default account is valid..."));
+			runTask(account->createCheckTask(m_session));
```

Comment:
Protip: Check the code before commenting. (in this case, check StandardTask::runTask)